File: halcyon-oauth2/src/main/java/com/marshal/halcyon/oauth2/controller/OauthClientDetailsController.java
package com.marshal.halcyon.oauth2.controller;

import com.marshal.halcyon.core.component.ResponseData;
import com.marshal.halcyon.core.controller.BaseController;
import com.marshal.halcyon.core.util.ResponseUtils;
import com.marshal.halcyon.oauth2.entity.OauthClientDetails;
import com.marshal.halcyon.oauth2.service.OauthClientDetailsService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * @auth: Marshal
 * @date: 2019/3/8
 * @desc:
 */
@RestController
@RequestMapping("/oauth2/clients")
public class OauthClientDetailsController extends BaseController {

    @Autowired
    OauthClientDetailsService oauthClientDetailsService;

    @RequestMapping("/query")
    public ResponseData query(@RequestBody OauthClientDetails condition,
                              @RequestParam int pageNum,
                              @RequestParam int pageSize) {
        List<OauthClientDetails> list = oauthClientDetailsService.select(condition, pageNum, pageSize);
        return new ResponseData(list);
    }

    @RequestMapping("/submit")
    public ResponseData save(@RequestBody OauthClientDetails oauthClientDetails) {
        if (!getValidator().isValid(oauthClientDetails)) {
            return new ResponseData(false, getValidator().getErrors(oauthClientDetails));
        }
        oauthClientDetailsService.updateByPrimaryKeySelective(oauthClientDetails);
        return ResponseUtils.responseOk("保存成功");
    }

    @RequestMapping("/remove")
    public ResponseData delete(@RequestParam("selectedIds") Long[] selectedIds) {
        oauthClientDetailsService.batchDelete(selectedIds);
        return ResponseUtils.responseOk("删除成功");
    }

    @RequestMapping("/queryById")
    public OauthClientDetails queryById(@RequestParam Long id) {
        return oauthClientDetailsService.selectByPrimaryKey(id);
    }

}

File: halcyon-web/src/main/resources/static/js/controller/oauth2/oauthClientDetailsController.js
//控制层
app.controller("oauthClientDetailsController", function ($scope, $controller, oauthClientDetailsService) {
    //继承
    $controller("baseController", {$scope: $scope});

    //初始化
    $scope.initForAdd = function () {
        $scope.oauthClientDetails = {};
    }
    $scope.initForUpdate = function (id) {
        oauthClientDetailsService.queryById(id).success(function (data) {
            $scope.oauthClientDetails = data;
        });
    }

    //crud
    //查询,初始化查询条件为空
    $scope.condition = {};
    $scope.query = function (pageNum, pageSize) {
        oauthClientDetailsService.query(pageNum, pageSize, $scope.condition).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //保存
    $scope.submit = function () {
        oauthClientDetailsService.submit($scope.oauthClientDetails).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    };
    //删除
    $scope.delete = function () {
        $scope.deleteConfirm(deleteRows);
    };

    function deleteRows() {
        oauthClientDetailsService.remove($scope.selectedIds).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }

    //获取下拉框数据
    //防止ajax异步导致为空，声明一个空的
    $scope.oauthClientDetailsOptions = {data: []};
    $scope.getOptions = function () {
        oauthClientDetailsService.getOptions().success(function (data) {
            $scope.oauthClientDetailsOptions = {data: data};
        });
    }

    $scope.empOptions = {data: []};
    $scope.getEmpOptions = function () {
        hrEmployeeService.getEmpOptions().success(function (data) {
            $scope.empOptions = {data: data};
        });
    }
});